datalake/src/test/java/com/sequenceiq/datalake/service/recovery/SdxRecoverySelectorServiceTest.java

```diff
@@ -0,0 +1,124 @@
+package com.sequenceiq.datalake.service.recovery;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.recovery.RecoveryStatus;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.service.resize.recovery.ResizeRecoveryService;
+import com.sequenceiq.datalake.service.upgrade.recovery.SdxUpgradeRecoveryService;
+import com.sequenceiq.sdx.api.model.SdxRecoverableResponse;
+import com.sequenceiq.sdx.api.model.SdxRecoveryRequest;
+import com.sequenceiq.sdx.api.model.SdxRecoveryResponse;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SdxRecoverySelectorServiceTest {
```

Comment:
pls cover this part of the service, too:

        String reasons = String.join(",", responseReasons);
        LOGGER.debug("Cluster is not in a recoverable state with message(s): {}", reasons);
        throw new BadRequestException(reasons);